doc/memory_usage.md

```diff
@@ -0,0 +1,133 @@
+Memory usage estimation for a module
```

Comment:
I think there are some overlaps with the https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/doc/memory_tune.md
Whereas I think both documents add good value, I wonder if we should rather merge them into a single high-quality doc about memory in general? Having said that, I don't mind keeping them separate for now, just a thought for the future.

Reply:
> I wonder if we should rather merge them into a single high-quality doc about memory in general?

ideally, maybe.